apps/financial-aid/web-veita/src/components/ApplicationHeader/ApplicationHeader.tsx

```diff
@@ -59,10 +62,24 @@ const ApplicationHeader = ({
       })
   }
 
+  const comingFromMyCases = (state: ApplicationState) =>
+    application &&
+    application.staff?.nationalId === admin?.staff?.nationalId &&
+    (state === ApplicationState.INPROGRESS ||
+      state === ApplicationState.DATANEEDED)
+
   const findPrevUrl = (
     state: ApplicationState,
-  ): React.SetStateAction<NavigationElement | undefined> => {
-    return navigationItems.find((i) => i.applicationState.includes(state))
+  ): SetStateAction<string | undefined> => {
+    if (comingFromMyCases(state)) {
+      return navigationItems.find((i) =>
+        i.applicationState.includes(ApplicationFiltersEnum.MYCASES),
+      )?.link
+    }
+
+    return navigationItems.find((i) =>
+      i.applicationState.includes(applicationStateToFilterEnum[state]),
+    )?.link
```

Comment:
Thinking are we not over engeering this? maybe just have user router.back here? Does the same as window.history.back()

Reply:
Let's try it out, not sure if it handles all use cases but if it doesn't then my vote is to throw it away.